common/es/src/main/java/io/apiman/common/es/util/EsConstants.java

```diff
@@ -244,30 +245,24 @@
     public static final String ES_MAPPING_TYPE_OBJECT = "object";
     public static final String ES_MAPPING_TYPE_TEXT = "text";
     // caches the es version read from the property file
-    private static String esVersion;
+    private static Properties esVersions;
 
     /**
      * Reads the elasticsearch version from the maven-generated properties file
      *
      * @return version the elasticsearch version
      */
-    public static String getEsVersion() {
-        if (esVersion != null) {
-            return esVersion;
-        } else {
-            java.io.InputStream is = EsConstants.class.getResourceAsStream("apiman-es.properties");
-            java.util.Properties p = new Properties();
-            String version;
+    public static Properties getEsVersion() {
+        if (esVersions == null) {
+            InputStream is = EsConstants.class.getResourceAsStream("apiman-es.properties");
+            Properties p = new Properties();
             try {
                 p.load(is);
             } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
If this is an unrecoverable error, then we should wrap the `IOException` in `UncheckedUIException` and rethrow it.

If it is ignored, then I suggest you either log it at trace level or do nothing.

Reply:
I am not totally sure if I understand you correctly. So this method is only used for tests, I think a `UncheckedUIException` is even more confusing?

Reply:
Changed it :)